contracts/upgradeability/CollectNFTProxy.sol

```diff
@@ -0,0 +1,19 @@
+pragma solidity 0.8.10;
+
+import {ILensHub} from '../interfaces/ILensHub.sol';
+import {Proxy} from '@openzeppelin/contracts/proxy/Proxy.sol';
+import {Address} from '@openzeppelin/contracts/utils/Address.sol';
+
+contract CollectNFTProxy is Proxy {
+    using Address for address;
+    address immutable HUB;
+
+    constructor(bytes memory data) {
+        HUB = msg.sender;
+        ILensHub(msg.sender).getCollectNFTImpl().functionDelegateCall(data);
+    }
+
+    function _implementation() internal view override returns (address) {
+        return ILensHub(HUB).getCollectNFTImpl();
+    }
+}
```

Comment:
As far as I understood we won't be using this, will we?

Reply:
Spot on, will remove!

Reply:
Removed in [be94b9f](https://github.com/aave/lens-protocol/pull/111/commits/be94b9ffe4f33283d38ea5f3def7922fb3cde5f8)